pkg/validation/config.go

```diff
@@ -11,30 +11,69 @@ import (
 	"github.com/openshift/ci-tools/pkg/api"
 )
 
+type configContext struct {
```

Comment:
Could we get a quick snippet on why this exists/what purpose it serves?

Reply:
Documentation, schmocumentation.  Added.